opty/tests/test_direct_collocation.py

```diff
@@ -14,6 +14,41 @@
                      _coo_matrix)
 
 
+def test_implicit_known_traj():
+
+    m, g, h = sym.symbols('m, g', real=True, nonnegative=True)
+    x, v, f, theta = mech.dynamicsymbols('x, v, f', real=True)
+    theta = sym.Function('theta')
+
+    states = (x, v)
+
+    eom = sm.Matrix([
+        m*v.diff() - f + m*g*sym.cos(theta(x)),
+        x.diff() - v,
+    ])
+
+    N = 20
+
+    def calc_theta(x):
```

Comment:
opty will need to know internally what to pass to `calc_theta`. This could be like the objective function and it only accepts the `free` array and it is  up to the user to do the calculations with the contents of `free` to ensure that an array is returned. This matches the objective function definitions.

An alternative is to make the symbolic term a function of the specific state, input, and parameter variables and then opty can know what specific values to pass to the function. This is more complicated to implement but maybe cleaner. I guess you'd need to be able to do things like `theta(x, v, m, g, f, t)` in general. But maybe we don't support explicit `t` (same as we don't yet in the EoMs.